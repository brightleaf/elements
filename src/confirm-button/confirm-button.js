import React from 'react'
import {
  Modal,
  useToggle,
  ModalCard,
  ModalCardBody,
  ModalCardFoot,
  ModalCardHead,
  Button,
  Field,
  Control,
} from '../'

const ConfirmButton = ({
  onConfirm,
  onCancel,
  question,
  confirmText,
  declineText,
  isDisabled,
  children,
  title,
}) => {
  const [modalShown, setModalShown] = useToggle(false)

  return (
    <>
      <Button
        disabled={isDisabled}
        isPrimary
        isLoading={modalShown}
        onClick={e => {
          setModalShown(true)
          e.preventDefault()
        }}
      >
        {children}
      </Button>
      <Modal
        includeTrigger={false}
        triggerText=""
        triggerFn={setModalShown}
        isActive={modalShown}
        ModalType={ModalCard}
        onCloseClick={onCancel}
      >
        <ModalCardHead title={title}></ModalCardHead>
        <ModalCardBody>{question}</ModalCardBody>
        <ModalCardFoot>
          <Field isGrouped>
            <Control>
              <Button
                isSuccess
                onClick={e => {
                  e.preventDefault()
                  onConfirm()
                  setModalShown(false)
                }}
              >
                {confirmText}
              </Button>
              <Button
                onClick={e => {
                  e.preventDefault()
                  onCancel()
                  setModalShown(false)
                }}
              >
                {declineText}
              </Button>
            </Control>
          </Field>
        </ModalCardFoot>
      </Modal>
    </>
  )
}

ConfirmButton.defaultProps = {
  confirmText: 'Ok',
  onCancel: () => {},
  onConfirm: () => {},
  declineText: 'Cancel',
  question: 'Are you sure?',
  title: 'Please confirm',
}
export { ConfirmButton }
export default ConfirmButton
