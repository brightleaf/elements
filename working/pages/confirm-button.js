/* eslint-disable spellcheck/spell-checker */
/* eslint-disable sonarjs/no-duplicate-string */
import React from 'react'
import { useStyleSheet } from '@brightleaf/react-hooks/lib/use-stylesheet'
import { useStyles } from '@brightleaf/react-hooks/lib/use-styles'
import Highlight from '../components/highlighter'
import {
  ConfirmButton,
  Box,
  Breadcrumb,
  BreadcrumbItem,
  Button,
  Buttons,
  Columns,
  Column,
  Container,
  FullColumn,
  Content,
  Hero,
  HeroBody,
  Title,
  SubTitle,
  Section,
} from '../../src'
import { Snippet } from '../components/snippet'
import { ElementsTabs } from '../components/tabs'
const Callout = ({ children }) => {
  return <div className="callout is-primary">{children}</div>
}
export default () => {
  useStyleSheet('https://fonts.googleapis.com/css?family=Open+Sans')
  useStyles(`
  html,
  body {
      font-family: 'Open Sans';
  }
  `)
  useStyleSheet('code.css')
  return (
    <Section>
      <div style={{ paddingTop: '3em', paddingBottom: '3em' }}>
        <Breadcrumb>
          <BreadcrumbItem>
            <a href="#/">Home</a>
          </BreadcrumbItem>
          <BreadcrumbItem>
            <a href="#/elements">Elements</a>
          </BreadcrumbItem>
          <BreadcrumbItem>
            <a href="#/buttons">Buttons</a>
          </BreadcrumbItem>
        </Breadcrumb>
      </div>

      <Hero isPrimary isBold>
        <HeroBody>
          <Title>Button</Title>
          <SubTitle as="p" is="4">
            The classic <strong>button</strong>, in different colors, sizes, and
            states
          </SubTitle>
        </HeroBody>
      </Hero>
      <div style={{ paddingTop: '3em', paddingBottom: '3em' }}>
        <ElementsTabs />
      </div>
      <FullColumn>
        <Content>
          <p>
            The <strong>button</strong> is an essential element of any design.
            It&apos;s meant to look and behave as an{' '}
            <strong>interactive</strong> element of your page.
          </p>
        </Content>
      </FullColumn>
      <Snippet>
        <Columns>
          <Column isHalf className="snippet-preview">
            <Container>
              <ConfirmButton>Button</ConfirmButton>
            </Container>
          </Column>
          <Column isHalf>
            <Highlight className="javascript" languages={['javascript']}>
              {`
import React from 'react'
import { ConfirmButton } from '@brightleaf/elements'

export default () => {
  return (
    <Container>
      <ConfirmButton>Button</ConfirmButton>
    </Container>
  )
}
`}
            </Highlight>
          </Column>
        </Columns>
      </Snippet>
      <FullColumn>
        <Content>
          <p>
            The <strong>button</strong> is an essential element of any design.
            It&apos;s meant to look and behave as an{' '}
            <strong>interactive</strong> element of your page.
          </p>
        </Content>
      </FullColumn>
      <Snippet>
        <Columns>
          <Column isHalf className="snippet-preview">
            <Container>
              <ConfirmButton
                question="Are you sure you want to do this?"
                onConfirm={e => {
                  console.log('confirm button click')
                }}
                onCancel={() => {
                  console.log('cancelled')
                }}
              >
                Do It
              </ConfirmButton>
            </Container>
          </Column>
          <Column isHalf>
            <Highlight className="javascript" languages={['javascript']}>
              {`
import React from 'react'
import { ConfirmButton } from '@brightleaf/elements'

export default () => {
  return (
    <Container>
      <ConfirmButton
        question="Are you sure you want to do this?"
        onConfirm={e => {
          console.log('confirm button click')
        }}
        onCancel={() => {
          console.log('cancelled')
        }}
      >
        Do It
      </ConfirmButton>
    </Container>
  )
}
`}
            </Highlight>
          </Column>
        </Columns>
      </Snippet>
    </Section>
  )
}
