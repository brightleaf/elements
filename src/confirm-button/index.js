import { ConfirmButton } from './confirm-button'

export { ConfirmButton }

export default ConfirmButton
