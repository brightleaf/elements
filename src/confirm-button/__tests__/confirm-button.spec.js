import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import { toHaveClass, toHaveAttribute } from '@testing-library/jest-dom'
import { ConfirmButton } from '../confirm-button'

expect.extend({ toHaveClass, toHaveAttribute })

describe('ConfirmButton components', () => {
  afterEach(cleanup)
  it('should render', () => {
    const { container } = render(<ConfirmButton>Button</ConfirmButton>)
    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('button')
  })
  it('should render a button', () => {
    const { container } = render(
      <ConfirmButton
        title="confirm"
        question="Are you sure you want to send the message?"
        onConfirm={e => {}}
        onCancel={() => {}}
      >
        Send
      </ConfirmButton>
    )
    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('button')
    expect(container.lastChild.firstChild).toHaveClass('modal')
  })
  it('should when the button is clicked add `is-active` class to modal', async () => {
    const { container } = render(
      <ConfirmButton
        title="confirm"
        question="Are you sure you want to send the message?"
        onConfirm={e => {}}
        onCancel={() => {}}
      >
        Send
      </ConfirmButton>
    )
    fireEvent.click(container.firstChild)
    const modal = await container.lastChild.firstChild
    expect(modal).toHaveClass('is-active')
  })
})
