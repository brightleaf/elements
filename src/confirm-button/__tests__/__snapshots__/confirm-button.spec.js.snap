// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmButton components should render 1`] = `
<div>
  <button
    class="button is-primary"
  >
    Button
  </button>
  <div>
    <div
      class="modal"
    >
      <div
        class="modal-background"
      />
      <div
        class="modal-card"
      >
        <header
          class="modal-card-head"
        >
          <p
            class="modal-card-title"
          >
            Please confirm
          </p>
          <button
            aria-label="close"
            class="delete"
          />
        </header>
        <section
          class="modal-card-body"
        >
          Are you sure?
        </section>
        <footer
          class="modal-card-foot"
        >
          <div
            class="field is-grouped"
          >
            <div
              class="control"
            >
              <button
                class="button is-success"
              >
                Ok
              </button>
              <button
                class="button"
              >
                Cancel
              </button>
            </div>
          </div>
        </footer>
      </div>
      <button
        aria-label="close"
        class="modal-close is-large"
      />
    </div>
  </div>
</div>
`;

exports[`ConfirmButton components should render a button 1`] = `
<div>
  <button
    class="button is-primary"
  >
    Send
  </button>
  <div>
    <div
      class="modal"
    >
      <div
        class="modal-background"
      />
      <div
        class="modal-card"
      >
        <header
          class="modal-card-head"
        >
          <p
            class="modal-card-title"
          >
            confirm
          </p>
          <button
            aria-label="close"
            class="delete"
          />
        </header>
        <section
          class="modal-card-body"
        >
          Are you sure you want to send the message?
        </section>
        <footer
          class="modal-card-foot"
        >
          <div
            class="field is-grouped"
          >
            <div
              class="control"
            >
              <button
                class="button is-success"
              >
                Ok
              </button>
              <button
                class="button"
              >
                Cancel
              </button>
            </div>
          </div>
        </footer>
      </div>
      <button
        aria-label="close"
        class="modal-close is-large"
      />
    </div>
  </div>
</div>
`;
